testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -2209,11 +2209,11 @@ public void testMergeSubqueries()
 
                 log.info("About to merge");
                 query(format("MERGE INTO %s t USING %s s", targetTable, sourceTable) +
-                        "    ON (t.nation_name = s.nation_name)" +
+                        "    ON (t.nation_name = s.nation_name AND (SELECT TRUE))" +
```

Comment:
we need tests exercising correlated queries, to ensure that

- scopes are correct and symbol bindings are correct
- decorrelation takes place and it's not totally screwed up (as it is easy with DELETEs and UPDATEs)

Reply:
I'd like some help constructing good correlated query examples for MERGE.

Reply:
I added tests for correlated queries in all places they were legal.

Reply:
@martint helped me figure out that the real problem in my second test was that correlated queries are not supported for RIGHT JOINs.

However, the test identified a real bug in MERGE: the `analyzeExpression` call to analyze the MERGE criterion must pass `CorrelationSupport.DISALLOWED` to ensure that the right error message issued in statement analysis.  This has been done.